tensorflow_probability/examples/keras_bayesian_neural_network.py

```diff
@@ -0,0 +1,360 @@
+        images, labels)
+    self.batch_size = batch_size
+
+  @staticmethod
+  def __generate_fake_data(num_images, num_classes):
+    '''
+    Generate fake data in the shape of the MNIST dataset for unittest.
+    Args:
+      num_images: Integer, the number of fake images to be generated.
+    Returns:
+      images: Numpy `array` representing the fake image data. The
+              shape of the array will be (num_images, 28, 28).
+      labels: Numpy `array` of integers, where each entry will be
+              assigned a unique integer.
+    '''
+    images = np.random.randint(low=0, high=256,
+                               size=(num_images, IMAGE_SHAPE[0],
+                                     IMAGE_SHAPE[1]))
+    labels = np.random.randint(low=0, high=num_classes,
+                               size=num_images)
+    return images, labels
+
+  @staticmethod
+  def __preprocessing(images, labels):
+    '''
+    Preprocessing for image and labels data.
+
+    Args:
+      images: Numpy `array` representing the image data.
+      labels: Numpy `array` representing the labels data (range 0-9).
+
+    Returns:
+      images: Numpy `array` representing the image data, normalized
+              and expanded for convolutional network input.
+      labels: Numpy `array` representing the labels data (range 0-9),
+              as one-hot (categorical) values.
+    '''
+    images = images/255.
+    images = np.expand_dims(images, axis=4)
+
+    labels = tf.keras.utils.to_categorical(labels)
+    return images, labels
+
+  def __len__(self):
+    return int(np.ceil(len(self.images)/self.batch_size))
+
+  def __getitem__(self, idx):
+    batch_x = self.images[idx*self.batch_size:(idx+1)*self.batch_size]
+    batch_y = self.labels[idx*self.batch_size:(idx+1)*self.batch_size]
+    return batch_x, batch_y
+
+
+def main(argv):
+  del argv  # unused
+  if tf.io.gfile.exists(FLAGS.model_dir):
+    tf.compat.v1.logging.warning(
+        "Warning: deleting old log directory at {}".format(FLAGS.model_dir))
+    tf.io.gfile.rmtree(FLAGS.model_dir)
+  tf.io.gfile.makedirs(FLAGS.model_dir)
+
+  if FLAGS.fake_data:
+    train_seq = MNISTSequence(batch_size=FLAGS.batch_size,
+                              size=NUM_TRAIN_EXAMPLES)
+    heldout_seq = MNISTSequence(batch_size=FLAGS.batch_size,
+                                size=NUM_HELDOUT_EXAMPLES)
+  else:
+    train_set, heldout_set = tf.keras.datasets.mnist.load_data()
+    train_seq = MNISTSequence(data=train_set, batch_size=FLAGS.batch_size)
+    heldout_seq = MNISTSequence(data=heldout_set, batch_size=FLAGS.batch_size)
+
+  model = create_model()
+
+  print(" ... Training convolutional neural network")
+  for epoch in range(FLAGS.num_epochs):
+    epoch_accuracy = []
+    for step, (batch_x, batch_y) in enumerate(train_seq):
+      # Eager mode returns a Tensor objest and not a scalar
+      # value for the loss, therefore only the mean accuracy
+      # is displayed.
```

Comment:
Eager mode vs graph mode per se shouldn't change the shapes of Tensors; I'm guessing this is a property of keras (or tf2 keras). I'd maybe remove this comment since it doesn't add all that much information that wasn't already present in the fact that you're calling `np.mean` below.